homeassistant/components/hassio.py

```diff
@@ -102,11 +141,21 @@ def async_setup(hass, config):
     def async_service_handler(service):
         """Handle service calls for HassIO."""
         api_command = MAP_SERVICE_API[service.service][0]
-        addon = service.data.get(ATTR_ADDON)
-        data = service.data[ATTR_INPUT] if ATTR_INPUT in service.data else None
+        addon = service.data.pop(ATTR_ADDON)
+        snapshot = service.data.pop(ATTR_SNAPSHOT)
+        payload = None
 
+        # Pass data to hass.io API
+        if ATTR_INPUT in service.data:
```

Comment:
This seems a weird if-else check. We're now sending more often data from services to Hass.io API. This seems to easy to get wrong when adding a new service.